config/apolloClient.ts

```diff
@@ -1,41 +1,131 @@
-import { ApolloClient, InMemoryCache, split, HttpLink } from "@apollo/client";
+import {
+  ApolloClient,
+  InMemoryCache,
+  HttpLink,
+  ApolloLink,
+} from "@apollo/client";
 import { getMainDefinition } from "@apollo/client/utilities";
 import { GraphQLWsLink } from "@apollo/client/link/subscriptions";
+import { TokenRefreshLink } from "apollo-link-token-refresh";
 import { createClient } from "graphql-ws";
+import decodeJWT from "jwt-decode";
+
+import { fetchRefreshToken } from "@/hooks/useAuth";
+import AuthTokensStore from "@/stores/AuthTokensStore";
+
+type headers = {
+  Authorization: string;
+};
 
 const HASURA_GRAPHQL_ENDPOINT = import.meta.env.VITE_HASURA_GRAPHQL_ENDPOINT;
 const HASURA_WS_ENDPOINT = import.meta.env.VITE_HASURA_WS_ENDPOINT;
 
-const httpLink = new HttpLink({
-  uri: HASURA_GRAPHQL_ENDPOINT,
-});
-
-const wsLink = new GraphQLWsLink(
-  createClient({
-    url: HASURA_WS_ENDPOINT,
-  })
-);
-
-// The split function takes three parameters:
-//
-// * A function that's called for each operation to execute
-// * The Link to use for an operation if the function returns a "truthy" value
-// * The Link to use for an operation if the function returns a "falsy" value
-const splitLink = split(
-  ({ query }) => {
-    const definition = getMainDefinition(query);
-    return (
-      definition.kind === "OperationDefinition" &&
-      definition.operation === "subscription"
-    );
-  },
-  wsLink,
-  httpLink
-);
+const getHeaders = () => {
+  const headers = {} as headers;
+  const { accessToken } = AuthTokensStore.getState();
+  if (accessToken) headers.Authorization = `Bearer ${accessToken}`;
+  return headers;
+};
+
+const operationIsSubscription = (operation: any): boolean => {
```

Comment:
let's not use the "any" type

Reply:
@DefK1lla how to change?
Please provide solution

Reply:
@ifokeev inside apollo there is an interface called Operation. 
![image](https://github.com/mlcraft-io/client-v2/assets/67697486/e27c2e30-cfe3-41a5-8189-32505d53a27a)
![image](https://github.com/mlcraft-io/client-v2/assets/67697486/841aa2ce-0010-4fdb-bb91-ab74571631d1)